### PR TITLE
[storage-queue] Modernized strings

### PR DESCRIPTION
**Pull request by wonhyeongseo**
# Description
closes https://github.com/Azure/azure-sdk-for-python/issues/22057

Hello, @jalauzon-msft! I am making a second pull request because I mistakedly deleted all of my changes to the first branch. As in the first, I removed unicode strings and replaced format methods with f-strings, may you please review them? Thank you. I hope you have a wonderful day!

P.S. May I change the `_shared` folder across the storage library one by one? I am aiming to do all of the libraries and sub libraries now.

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md##building-and-testing)
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

**Review by mccoyp — COMMENTED** [review]
This is great, thank you @wonhyeongseo! I'll defer to @jalauzon-msft to approve or comment, but this looks good to me 🙂

File: sdk/storage/azure-storage-queue/azure/storage/queue/_shared/authentication.py [hunk 1]
```diff
@@ -38,15 +38,7 @@ def _wrap_exception(ex, desired_type):
     msg = ""
     if ex.args:
         msg = ex.args[0]
-    if sys.version_info >= (3,):
```
Comment (jalauzon-msft) [inline sdk/storage/azure-storage-queue/azure/storage/queue/_shared/authentication.py:41]: One lint error in this file now that you've removed the use of `sys`.
`azure/storage/queue/_shared/authentication.py:8: [W0611(unused-import), ] Unused import sys`

**Review by jalauzon-msft — APPROVED** [review]
Thanks @wonhyeongseo! This looks good to me! Just one lint error that I left a comment for that would be great if you could address.

As for `_shared`, it is okay to address them in a separate PR as long as all packages are updated eventually. Feel free to just update `_shared` in other packages if you don't want to change all folders in other packages (blob is pretty big :) )

File: sdk/storage/azure-storage-queue/azure/storage/queue/_shared/authentication.py (hunk 1 above)
Comment (wonhyeongseo) [reply sdk/storage/azure-storage-queue/azure/storage/queue/_shared/authentication.py:41]: Hello @jalauzon-msft ! Thank you so much for the review! I've removed the import statement for `sys` in a new commit. Please let me know if I can improve on the PR. 

I love a good challenge, so modernizing the entire storage blob will be a treat 😆 
In addition to that, I want to PR some more technical issues, but don't know where to start really. May you please refer me to some beginner-friendly issues? Thanks and have a wonderful day!

Comment (jalauzon-msft) [reply sdk/storage/azure-storage-queue/azure/storage/queue/_shared/authentication.py:41]: Hmm, I'm not entirely sure of any issues that we have that would be beginner-friendly but let me look through the backlog and get back to you. Thanks.

**Comment by jalauzon-msft** [comment]
Thanks @wonhyeongseo! I've merged this change.

**Pull request merged**